controllers/awsmachine_controller.go

```diff
@@ -745,7 +754,9 @@ func (r *AWSMachineReconciler) cloudInitUserData(machineScope *scope.MachineScop
 	return encryptedCloudInit, nil
 }
 
-func (r *AWSMachineReconciler) ignitionUserData(scope *scope.MachineScope, objectStoreSvc services.ObjectStoreInterface, userData []byte) ([]byte, error) {
+// ignitionUserDataWithS3 creates an S3 bucket object and stores user data in it,
+// then returns the config to instruct ignition how to pull the user data from S3.
+func (r *AWSMachineReconciler) ignitionUserDataWithS3(scope *scope.MachineScope, objectStoreSvc services.ObjectStoreInterface, userData []byte) ([]byte, error) {
 	if objectStoreSvc == nil {
 		return nil, errors.New("object store service not available")
```

Comment:
```suggestion
		return nil, errors.New("using Ignition requires an object storage by default, this can be se in `AWSCluster.Spec.Ignition.S3Bucket`, or to use EC2 user-data set `AWSMachine.Spec. .Ignition.UseUnencryptedUserData`")
```